common/utils/yaml_test.go

```diff
@@ -80,10 +79,42 @@ func TestMergeYAMLWithSlices(t *testing.T) {
 
 			var gotYAML map[string]interface{}
 			var expectedYAML map[string]interface{}
-
-			gotMap := yaml.Unmarshal([]byte(got), &gotYAML)
-			expectedMap := yaml.Unmarshal([]byte(tc.expectedResult), &expectedYAML)
-			assert.Equal(t, gotMap, expectedMap, "expected %v, got %v", expectedMap, gotMap)
+			err = yaml.Unmarshal([]byte(got), &gotYAML)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you tried https://pkg.go.dev/github.com/stretchr/testify/assert#Assertions.YAMLEq?
It would be:
```suggestion
			assert.YAMLEq(t, tc.expectedResult, got)
```

Reply:
I just tried, and that does not work for our use case. Because in YAML
```
tags:
  - a
  - b
```
and
```
tags:
  - b
  - a
 ```
are not considered equal. Because they are list and the order is important. In our case we do not care about the order, so we cannot really use `assert.YAMLEq` directly

But using `assert.ElementsMatch` already simplify the code! Thanks!